src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int LENGTH_STORAGE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size = 0;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[LENGTH_STORAGE];
+        values = (V[]) new Object[LENGTH_STORAGE];
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i <= size; i++) {
+            if (values[i] != null && (key == keys[i]) || (keys[i] != null
+                    && keys[i].equals(key))
+                    || (keys[i] == null && values[i] == null)) {
```

Comment:
```suggestion
            if (key == key[i] || key != null && key.equals(keys[i])) {
```
no need to check if the value is not null, it may be